addons/point_of_sale/models/pos_order.py

```diff
@@ -278,6 +278,10 @@ def insert_data(data_type, values):
                     name = name + ' (' + line.notice + ')'
 
                 # Create a move for the line for the order line
+                # Just like for invoices, a group of taxes must be present on this base line
+                # As well as its children
+                base_line_tax_ids = (line.tax_ids_after_fiscal_position +
+                    line.tax_ids_after_fiscal_position.mapped('children_tax_ids').filtered(lambda tax: tax.type_tax_use in ['sale', 'none']))
```

Comment:
although it will work for the first level of children, it won't for groups of depth > 1

I think you need to use a recursive function